src/System.Text.RegularExpressions/tests/RegexCompilationHelper.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace System.Text.RegularExpressions.Tests
+{
+    public class RegexCompilationHelper
+    {
+        public static IEnumerable<object[]> TransformRegexOptions(string testDataMethodName, int regexOptionsArrayIndex)
+        {
+            // On Uap or NetNative the compiled feature isn't currently enabled,
+            // therefore we don't need the additional test data.
+            if (PlatformDetection.IsUap)
+            {
+                return Enumerable.Empty<object[]>();
+            }
+
+            int splitIndex = testDataMethodName.IndexOf('.');
+            if (splitIndex > 0)
+            {
+                string typeName = testDataMethodName.Substring(0, splitIndex);
```

Comment:
I can't see how this block is reached. All the methods seem to have no dots.

Reply:
Hmm I guess I over-engineered ;)